starknet_devnet/blueprints/feeder_gateway.py

```diff
@@ -210,6 +211,18 @@ async def get_transaction_status():
     tx_status = await state.starknet_wrapper.transactions.get_transaction_status(
         transaction_hash
     )
+
+    # In case of blocks-on-demand feature tx_status should be received
+    if (
+        state.starknet_wrapper.config.blocks_on_demand
+        and tx_status["tx_status"] == TransactionStatus.NOT_RECEIVED.name
+        and any(
+            hex(transaction.transaction_hash) == transaction_hash
+            for transaction in state.starknet_wrapper.mempool
+        )
+    ):
+        tx_status["tx_status"] = TransactionStatus.RECEIVED.name
```

Comment:
TBH this looks hacky. Why can't we just store them as RECEIVED, and then during block creation just change the status of each tx to ACCEPTED_ON_L2

Reply:
Ok, let's discuss it later on a call.